pkg/ccl/backupccl/backupinfo/manifest_handling.go

```diff
@@ -911,6 +912,28 @@ func LoadSQLDescsFromBackupsAtTime(
 	return allDescs, lastBackupManifest
 }
 
+// FindLatestReIntroductionByTable finds the endtime of the latest incremental
+// backup that reintroduced each backed up table, as of restore time. Note: this
+// function assumes that each introduced span in the manifest covers a single table.
+func FindLatestReIntroductionByTable(
+	manifests []backuppb.BackupManifest, codec keys.SQLCodec, asOf hlc.Timestamp,
+) (map[descpb.ID]hlc.Timestamp, error) {
+	latestReIntro := make(map[descpb.ID]hlc.Timestamp)
+	for _, b := range manifests {
+		if !asOf.IsEmpty() && asOf.Less(b.StartTime) {
+			break
+		}
+		for _, sp := range b.ReintroducedSpans {
```

Comment:
wouldn't being in _introduced_ spans (as opposed to reintroduced) as of layer X be sufficient to say that we shouldn't be restoring the span from files from layer prior to X? do we really need to specifically track Reintroduced separately?

Reply:
not necessarily -- a new index on an old table gets included in `IntroducedSpans`, but we certainly don't want to elide all data from old backups associated with that table. See this [comment](https://github.com/cockroachdb/cockroach/pull/87312#issuecomment-1242545077) for more detail and an example.

You then might wonder, "why not keep a `map[table_index]hlc.timestamp` then?" Well, that leads to problems too, as explained in this [thread.](https://github.com/cockroachdb/cockroach/pull/87312#discussion_r967031256) . I'll lay out the alternatives considered into updated commit message.

Reply:
Isn't there a distinction here between putting just the index span into introduced span versus putting the whole table span in it?

Reply:
Aha! The first n spans in backupManifest.IntroducedSpans are the new index spans. On the restore side, i think we can identify the new spans by calling:

`newSpans = filterSpans(latestBackup.Spans, prevBackups[len(prevBackups)-1].Spans)`

, remove them from IntroducedSpans, and get the reIntroducedSpans. 

This means we don't need to modify the backup manifest. Brilliant!

Reply:
no longer relevant.